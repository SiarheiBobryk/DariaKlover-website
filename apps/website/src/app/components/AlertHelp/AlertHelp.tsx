import { MouseEventHandler, ReactNode, Ref, forwardRef } from 'react';

import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

import LinkEmail from '../LinkEmail';
import LinkSocial from '../LinkSocial';

export interface AlertHelpProps {
  title?: string;
  children?: ReactNode;
  open?: boolean;
  onClose?: MouseEventHandler<HTMLButtonElement>;
  severity?: 'success' | 'info' | 'warning' | 'error';
}

const AlertHelp = forwardRef(function AlertHelp(props: AlertHelpProps, ref: Ref<HTMLDivElement>) {
  const {
    title = 'Связаться с нами',
    children = (
      <>
        Если у вас остались вопросы или вы продолжаете испытывать сложности в работе с сайтом, то не стесняйтесь и
        свяжитесь с нами через <LinkSocial social="telegram" />, <LinkSocial social="whatsapp" /> или по <LinkEmail />.
        Мы обязательно поможем.
      </>
    ),
    open,
    onClose,
    ...other
  } = props;

  return (
    <Alert ref={ref} severity="info" variant="outlined" {...other}>
      <AlertTitle>{title}</AlertTitle>
      {children}
    </Alert>
  );
});

export default AlertHelp;
