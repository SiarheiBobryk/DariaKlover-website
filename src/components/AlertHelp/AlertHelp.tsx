import * as React from 'react';

import Alert from '@mui/material/Alert';
import AlertTitle from '@mui/material/AlertTitle';

import LinkSocial from '../LinkSocial';
import LinkEmail from '../LinkEmail';

export interface AlertHelpProps {
  children?: React.ReactNode;
  open?: boolean;
  onClose?: React.MouseEventHandler<HTMLButtonElement>;
  severity?: 'success' | 'info' | 'warning' | 'error';
}

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const AlertHelp = React.forwardRef(function AlertHelp(props: AlertHelpProps, ref: React.Ref<any>) {
  const { children, open, onClose, ...other } = props;

  return (
    <Alert ref={ref} severity="info" variant="outlined" {...other}>
      <AlertTitle>Связаться с нами</AlertTitle>
      Если у вас остались вопросы или вы продолжаете испытывать сложности в работе с сайтом, то не стесняйтесь и
      свяжитесь с нами через <LinkSocial social="telegram" />, <LinkSocial social="whatsapp" /> или по <LinkEmail />. Мы
      обязательно поможем.
    </Alert>
  );
});

export default AlertHelp;
