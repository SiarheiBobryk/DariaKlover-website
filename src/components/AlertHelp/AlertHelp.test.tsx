import * as React from 'react';

import { axe, toHaveNoViolations } from 'jest-axe';
import { render, RenderResult, screen } from '@testing-library/react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';

import AlertHelp from './AlertHelp';

expect.extend(toHaveNoViolations);

describe('<AlertHelp />', () => {
  it('should match a11y standards without props', async () => {
    const { container }: RenderResult = render(<AlertHelp />);

    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('should forward ref', () => {
    const ref = React.createRef();
    render(<AlertHelp ref={ref} data-testid="AlertHelp" />);
    expect(screen.getByTestId('AlertHelp')).toBe(ref.current);
  });

  it('should render correctly', () => {
    const alertHelp: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer.create(<AlertHelp />).toJSON();
    expect(alertHelp).toMatchSnapshot();
  });
});
