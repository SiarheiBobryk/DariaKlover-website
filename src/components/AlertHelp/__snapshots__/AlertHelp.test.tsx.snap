// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertHelp /> should render correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardInfo MuiAlert-standard css-12mvehg-MuiPaper-root-MuiAlert-root"
  role="alert"
>
  <div
    className="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
      data-testid="InfoOutlinedIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
      />
    </svg>
  </div>
  <div
    className="MuiAlert-message css-1pxa9xg-MuiAlert-message"
  >
    <div
      className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
    >
      Связаться с нами
    </div>
    Если у вас остались вопросы или вы продолжаете испытывать сложности в работе с сайтом, то не стесняйтесь и свяжитесь с нами через 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://t.me/dariaklover"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      target="_blank"
    >
      Telegram
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
    , 
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="https://wa.me/+375447841051"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      target="_blank"
    >
      WhatsApp
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
     или по
     
    <a
      className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
      href="mailto: dariaklover@gmail.com"
      onBlur={[Function]}
      onFocus={[Function]}
      rel="noopener"
      target="_blank"
    >
      электронной почте (
      <i>
        Email
      </i>
      )
       
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1xzgpfl-MuiSvgIcon-root"
        data-testid="OpenInNewIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
        />
      </svg>
    </a>
    . Мы обязательно поможем.
  </div>
</div>
`;
